File: VerticalLoad.cs
namespace Master_Thesis;

class Load
{
    private double _verticalLoad;

    // Calculating total value of characteristic vertical load acting on the structure: concrete dead load, finishing layers on each floor, live load on each floor
    public Load(double width, double length, double tFloor, double gLoad, double qLoad, List<Wall> walls, double hFloor, double tWall, double columnCs, double columnNr, double floorNr)
    {
        const int densityOfConcrete = 25000;
        var floorDeadLoad = (width * length) * (gLoad + densityOfConcrete * tFloor) * floorNr;
        var floorLiveLoad = (width * length * qLoad) * floorNr;

        double lengthWalls = 0;

        foreach (Wall wall in walls)
        {
            lengthWalls += wall.Length();
        }

        var wallsDeadLoad = (lengthWalls * hFloor * tWall * densityOfConcrete) * floorNr;
        var columnDeadLoad = (hFloor * columnCs * columnNr * densityOfConcrete) * floorNr;

        _verticalLoad = floorDeadLoad + wallsDeadLoad + columnDeadLoad + floorLiveLoad;
    }

    // Calling final value of the load
    public double GetVerticalLoad()
    { return _verticalLoad; }
    
}

File: Python/Vertical_Load.py
class Load:
    _verticalLoad = None

    # Calculating total value of characteristic vertical load acting on the structure: concrete dead load, finishing layers on each floor, live load on each floor
    def __init__(self, width, length, tFloor, gLoad, qLoad, walls, hFloor, columnCs, columnNr, floorNr):
        densityOfConcrete = 25000
        floorDeadLoad = (width * length) * (gLoad + densityOfConcrete * tFloor) * floorNr
        floorLiveLoad = (width * length * qLoad) * floorNr

        wallsDeadLoad = 0

        for wall in walls:
            wallsDeadLoad += wall.length * wall.thickness * hFloor * densityOfConcrete * floorNr

        columnDeadLoad = (hFloor * columnCs * columnNr * densityOfConcrete) * floorNr

        self._verticalLoad = floorDeadLoad + wallsDeadLoad + columnDeadLoad + floorLiveLoad

    # Calling final value of the load
    def GetVerticalLoad(self):
        return self._verticalLoad
